File: Java/src/main/java/QuestionBank/Solution2001_2100/Solution2028.java
package QuestionBank.Solution2001_2100;
/**
 * @author windows
 */
public class Solution2028 {
    public int[] missingRolls(int[] rolls, int mean, int n) {
        int sum = 0;
        int m = rolls.length;
        for (int i = 0; i < m; i++) {
            sum += rolls[i];
        }
        int total = (m + n) * mean;
        int sub = total - sum;
        if (sub > n * 6 || sub < n){
            return new int[0];
        }
        int a = sub / n;
        int b = sub % n;
        int[] res = new int[n];
        for (int i = 0; i < n; i++) {
            if (b > 0){
                res[i]++;
                b--;
            }
            res[i] += a;
        }
        return res;
    }
    
    
    public static void main(String[] args) {
        Solution2028 solution2028 = new Solution2028();
    }
}

File: C++/src/Solution2001_2100/Solution2028.cpp
//
// Created by windows on 2023/3/2.
//

#include "Solution2028.h"
#include <functional>
#include <iostream>
#include <vector>
#include <unordered_set>
#include <unordered_map>
#include <string>
using namespace std;

vector<int> missingRolls(vector<int>& rolls, int mean, int n) {
  int sum = 0, m = rolls.size();
  for (int i = 0; i < m; ++i) {
    sum += rolls[i];
  }
  int total = mean * (n + m) - sum;
  if (total < n || total > n * 6) {
    return {};
  }
  vector<int> ans(n);
  for (int i = 0; i < n; ++i) {
    ans[i] = total / n;
  }
  int x = total - total / n * n;
  for (int i = 0; i < x; ++i) {
    ans[i]++;
  }
  return ans;
}